cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaCluster.java

```diff
@@ -1594,7 +1596,13 @@ private Container createContainer(ImagePullPolicy imagePullPolicy, KafkaPool poo
      */
     private  List<EnvVar> getEnvVars(KafkaPool pool) {
         List<EnvVar> varList = new ArrayList<>();
-        varList.add(ContainerUtils.createEnvVar(ENV_VAR_KAFKA_METRICS_ENABLED, String.valueOf(metrics.isEnabled())));
+        if (jmxExporterMetrics.isEnabled()) {
+            varList.add(ContainerUtils.createEnvVar(ENV_VAR_KAFKA_METRICS_ENABLED, String.valueOf(jmxExporterMetrics.isEnabled())));
+        } else if (strimziReporterMetrics.isEnabled()) {
+            varList.add(ContainerUtils.createEnvVar(ENV_VAR_KAFKA_METRICS_ENABLED, String.valueOf(strimziReporterMetrics.isEnabled())));
+        } else {
+            varList.add(ContainerUtils.createEnvVar(ENV_VAR_KAFKA_METRICS_ENABLED, "false"));
+        }
```

Comment:
Both `jmxExporterMetrics` and `strimziReporterMetrics` can be null. As a hint, this can be greatly simplified by using the ternary operator.

Reply:
And this?i

 ```
List<EnvVar> varList = new ArrayList<>();
        varList.add(ContainerUtils.createEnvVar(
                ENV_VAR_KAFKA_JMX_EXPORTER_ENABLED,
                jmxExporterMetrics != null && jmxExporterMetrics.isEnabled()
                        ? "true"
                        : strimziReporterMetrics != null && strimziReporterMetrics.isEnabled()
                        ? "true"
                        : "false"
        ));
```

Reply:
This can still be simplified and made easier to read like this:

```sh
String jmxMetricsEnabled = jmxExporterMetrics != null && jmxExporterMetrics.isEnabled() ? Boolean.TRUE.toString() : Boolean.FALSE.toString();
        varList.add(ContainerUtils.createEnvVar(ENV_VAR_KAFKA_JMX_EXPORTER_ENABLED, jmxMetricsEnabled));
```